services/api/src/api/jwt_token.py

```diff
@@ -0,0 +1,161 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 The HuggingFace Authors.
+
+from typing import Any, Optional, Union
+
+import jwt
+import requests
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric.ec import (
+    EllipticCurvePrivateKey,
+    EllipticCurvePublicKey,
+)
+from cryptography.hazmat.primitives.asymmetric.ed448 import (
+    Ed448PrivateKey,
+    Ed448PublicKey,
+)
+from cryptography.hazmat.primitives.asymmetric.ed25519 import (
+    Ed25519PrivateKey,
+    Ed25519PublicKey,
+)
+from cryptography.hazmat.primitives.asymmetric.rsa import RSAPrivateKey, RSAPublicKey
+from jsonschema import ValidationError, validate
+from jwt.algorithms import (
+    ECAlgorithm,
+    HMACAlgorithm,
+    OKPAlgorithm,
+    RSAAlgorithm,
+    RSAPSSAlgorithm,
+)
+
+from api.utils import JWKError
+
+ASYMMETRIC_ALGORITHMS = (ECAlgorithm, OKPAlgorithm, RSAAlgorithm, RSAPSSAlgorithm)
+SYMMETRIC_ALGORITHMS = (HMACAlgorithm,)
+
+
+def is_public_key(
+    key: Union[
+        EllipticCurvePublicKey,
+        EllipticCurvePrivateKey,
+        Ed25519PublicKey,
+        Ed448PublicKey,
+        Ed25519PrivateKey,
+        Ed448PrivateKey,
+        RSAPrivateKey,
+        RSAPublicKey,
+    ]
+) -> Union[EllipticCurvePublicKey, Ed25519PublicKey, Ed448PublicKey, RSAPublicKey]:
```

Comment:
Nitpick: This function does not return a `Union[...]` but a `bool`

Reply:
🤦